zebra/zapi_msg.c

```diff
@@ -1064,7 +1065,7 @@ static void zread_rnh_register(ZAPI_HANDLER_ARGS)
 				p.family);
 			return;
 		}
-		rnh = zebra_add_rnh(&p, zvrf_id(zvrf), type);
+		rnh = zebra_add_rnh(&p, zvrf_id(zvrf), type, &exist);
```

Comment:
nit: `zebra_add_rnh()` might return NULL in the case of an error and we're not checking this possibility here (this is a pre-existing problem). CS might complain that the `exists` variable might be used uninitialized in this case. Checking if `rnh` is NULL or not would solve both problems. Other than that the changes look good to me.

Reply:
yeah it is a bit of a pre-existing problem( but only going to happen when someone moves around initialization ) and the code would quickly crash too due to rnh deref.  I'll make it behave a bit better though.